desktop/mac/permission-requirements.md

```diff
@@ -7,15 +7,85 @@ redirect_from:
+
+The docker binaries are installed by default in `/Applications/Docker.app/Contents/Resources/bin`. Docker Desktop ensures the `docker` CLI is on the user’s PATH without having to reconfigure shells, log out then log back in for example. As on most systems, `/usr/local/bin` is by default in user's PATH, Docker Desktop creates symlinks for all docker binaries in it.
+
+Installing symlinks in `/usr/local/bin` is a privileged configuration Docker Desktop performed on the first startup. Docker Desktop checks if symlinks exists and takes the following actions: 
+- creates the symlinks without the admin prompt if `/usr/local/bin` is writable by unprivileged users
+- triggers an admin prompt for user to authorize the creation of symlinks in `/usr/local/bin`. If user authorizes, symlinks to docker binaries are created in `/usr/local/bin`.
+If user rejects the prompt, not willing to run configurations requiring elevated privileges or due to not having admin rights on its machine, Docker Desktop falls back to create the symlinks in `~/.docker/bin` and edit user's shell profile to ensure this location is in user's PATH. This requires all open shells to be reloaded. 
+The rejection is recorded for future runs to avoid prompting the user again.
+For any failure to ensure binaries are on user's PATH, the user may need to add manually to its PATH the `/Applications/Docker.app/Contents/Resources/bin` or use full path to docker binaries.
+
+A particular case is the installation of the `/var/run/docker.sock` symlink. Creating this symlink ensures various docker clients relying on the default docker socket path to work without additional changes. As the `/var/run` is mounted as a tmpfs, its content is deleted on restart, symlink to docker socket included.
+To ensure the docker socket exists after restart, Docker Desktop sets up a launchd startup task that creates the symlink by running `ln -s -f /Users/<user>/.docker/run/docker.sock /var/run/docker.sock`. This ensures, the user is not prompted all the time on startup to create the symlink. If user rejects the prompt, the symlink and the startup task is not created and user may have to explicitly set the DOCKER_HOST to `/Users/<user>/.docker/run/docker.sock` in the clients it is using. The docker CLI relies on the current context to retrieve the socket path, current context is set to `desktop-linux` on Docker Desktop startup.
+
+### Binding privileged ports
+
+When running container that require privileged ports binding, Docker Desktop first attempts to bind it directly as unprivileged process. In case of failure due to the OS preventing it, Docker Desktop checks if the `com.docker.vmnetd` privileged helper process is running to bind the privileged port through it. If the privileged helper process is not running, Docker Desktop prompts the user for authorization to run it under [launchd](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html).
```

Comment:
```suggestion
When running a container that requires privileged ports binding, Docker Desktop first attempts to bind it directly as an unprivileged process. In case of failure due to the OS preventing it, Docker Desktop checks if the `com.docker.vmnetd` privileged helper process is running to bind the privileged port through it. If the privileged helper process is not running, Docker Desktop prompts the user for authorization to run it under [launchd](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html).
```